content/specs/data-model.md

```diff
@@ -0,0 +1,146 @@
+
+Lexicon string fields can have additional `format` type information associated with them for validation, but as with other validation constraints this information is not available without the Lexicon itself.
+
+### Nullable and False-y
+
+In the atproto data model there is a semantic difference between explicitly setting an map field to `null` and not including the field at all. Both JSON and CBOR have the same distinction.
+
+Null or missing fields are also distinct from "false-y" value like `false` (for booleans), `0` (for integers), empty lists, or empty objects.
+
+### Why No Floats?
+
+IPLD, CBOR, and JSON all natively support floating point numbers, so why does atproto go out of the way to disallow them?
+
+The IPLD specification describes some of the complexities and sharp edges when working with floats in a content-addressable world. In short, de-serializing in to machine-native format, then later re-encoding, is not always consistent. This is definitely true for special values and corner-cases, but can even be true with "normal" float values on less-common architectures.
+
+It may be possible to come up with rules to ensure reliable round-trip encoding of floats in the future, but for now we disallow floats.
+
+If you have a use-case where integers can not be substituted for floats, we recommend encoding the floats as strings or even bytes. This provides a safe default round-trip representation.
+
+## `blob` Type
+
+References to "blobs" (arbitrary files) have a consistent format in atproto, and can be detected and processed without access to any specific Lexicon. That is, it is possible to parse nodes and extract any blob references without knowing the schema.
+
+Blob nodes are maps with following fields:
+
+- `$type` (string, required): fixed value `blob`. Note that this is not a valid NSID.
+- `ref` (link, required): CID reference to blob, with multicodec type `raw`. In JSON, encoded as a `$link` object as usual
+- `mimeType` (string, required, not empty): content type of blob. `application/octet-stream` if not known
+- `size` (integer, required, positive, non-zero): length of blob in bytes
+
+There is also a deprecated legacy blob format, with some records in the wild still containing blob references in this format:
+
+- `cid` (string, required): a CID in *string* format, not *link* format
+- `mimeType` (string, required, not empty): same as `mimeType` above
+
+Note that the legacy format has no `$type` and can only be parsed for known Lexicons. Implementations should not throw errors when encountering the old format, but should never write them, and it is acceptable to only partially support them.
+
+## JSON Representation
+
+atproto uses it's own conventions for JSON, instead of using DAG-JSON directly. Normalizations like key sorting are also not required or enforced: only DAG-JSON is used as a byte-reproducible representation.
```

Comment:
```suggestion
atproto uses its own conventions for JSON, instead of using DAG-JSON directly. Normalizations like key sorting are also not required or enforced: only DAG-JSON is used as a byte-reproducible representation.
```